packages/control-property-editor/src/panels/changes/GenericProperty.tsx

```diff
@@ -3,15 +3,22 @@ import React from 'react';
 import styles from './DisplayAsIcon.module.scss';
 
 interface GenericPropertyProps {
-    value: string | number | boolean | object;
+    value?: string | number | boolean;
     label: string;
 }
 
 const GenericProperty: React.FC<GenericPropertyProps> = ({ value, label }) => {
+    const convertedLabel = convertCamelCaseToPascalCase(label);
     return (
         <>
-            <span className={styles.genericPropLabel}>{convertCamelCaseToPascalCase(label)}:</span>
-            <span className={styles.genericPropValue}>{typeof value === 'object' ? '[Object]' : value}</span>
+            <span className={styles.genericPropLabel} title={convertedLabel}>
+                {convertedLabel}:
+            </span>
+            {(value !== undefined || value !== null) && (
+                <span className={styles.genericPropValue} title={value?.toString()}>
```

Comment:
The value check doesn't look right and we shouldn't need optional chaining operator later on.
```suggestion
            {value !== undefined && value !== null && (
                <span className={styles.genericPropValue} title={value.toString()}>
```